cpp/run3ScoutingLooper.C

```diff
@@ -0,0 +1,587 @@
+
+      // Muon selection
+      auto mus = getObject<std::vector<Run3ScoutingMuon>>(ev, "hltScoutingMuonPacker");
+      auto jets = getObject<std::vector<Run3ScoutingPFJet>>(ev, "hltScoutingPFPacker");
+      auto pfs = getObject<std::vector<Run3ScoutingParticle>>(ev, "hltScoutingPFPacker");
+      Muons.clear();
+      unsigned int nMus = mus.size();
+      nMuonAssoc=0;
+
+      for (unsigned int iMu=0; iMu<nMus; ++iMu) {
+        auto mu = mus[iMu];
+        std::vector<int> matchedAndSelVtxIdxs;
+        for (auto matchedVtxIdx : mu.vtxIndx()) {
+          for (auto selVtxIdx : SVs.index) {
+            if (matchedVtxIdx==selVtxIdx)
+              matchedAndSelVtxIdxs.push_back(matchedVtxIdx);
+          }
+        }
+        if (matchedAndSelVtxIdxs.size() < 1) // Require muon to be associated to at least one of the selected SVs
+          continue;
+        nMuonAssoc++; // Count muon associated to at least one selected SV
+        if (!(mu.pt()>3.0 && fabs(mu.eta())<2.4))
+          continue;
+        Muons.vtxIdxs.push_back(matchedAndSelVtxIdxs);
+        Muons.saHits.push_back(mu.nValidStandAloneMuonHits());
+        Muons.saMatchedStats.push_back(mu.nStandAloneMuonMatchedStations());
+        Muons.muHits.push_back(mu.nValidRecoMuonHits());
+        Muons.muChambs.push_back(mu.nRecoMuonChambers());
+        Muons.muCSCDT.push_back(mu.nRecoMuonChambersCSCorDT());
+        Muons.muMatch.push_back(mu.nRecoMuonMatches());
+        Muons.muMatchedStats.push_back(mu.nRecoMuonMatchedStations());
+        Muons.muExpMatchedStats.push_back(mu.nRecoMuonExpectedMatchedStations());
+        Muons.muMatchedRPC.push_back(mu.nRecoMuonMatchedRPCLayers());
+        Muons.pixHits.push_back(mu.nValidPixelHits());
+        Muons.stripHits.push_back(mu.nValidStripHits());
+        Muons.pixLayers.push_back(mu.nPixelLayersWithMeasurement());
+        Muons.trkLayers.push_back(mu.nTrackerLayersWithMeasurement());
+        Muons.pt.push_back(mu.pt());
+        Muons.eta.push_back(mu.eta());
+        Muons.phi.push_back(mu.phi());
+        Muons.ch.push_back(mu.charge());
+        Muons.chi2ndof.push_back(mu.normalizedChi2());
+        Muons.ecalIso.push_back(mu.ecalIso());
+        Muons.hcalIso.push_back(mu.hcalIso());
+        Muons.trackIso.push_back(mu.trackIso());
+        Muons.ecalRelIso.push_back(mu.ecalIso()/mu.pt());
+        Muons.hcalRelIso.push_back(mu.hcalIso()/mu.pt());
+        Muons.trackRelIso.push_back(mu.trackIso()/mu.pt());
+        Muons.dxy.push_back(mu.trk_dxy());
+        Muons.dxye.push_back(mu.trk_dxyError());
+        Muons.dz.push_back(mu.trk_dz());
+        Muons.dze.push_back(mu.trk_dzError());
+        Muons.dxysig.push_back(mu.trk_dxy()/mu.trk_dxyError());
+        Muons.dzsig.push_back(mu.trk_dz()/mu.trk_dzError());
+
+        TLorentzVector muVec; muVec.SetPtEtaPhiM(mu.pt(), mu.eta(), mu.phi(), MUON_MASS);
+        Muons.vec.push_back(muVec);
+
+        if (isGlobalMuon(mu.type()) && isTrackerMuon(mu.type()))
```

Comment:
As this is a vector of strings, should probably assign directly the corresponding string.
However, it may be best to just store three booleans in place of "type": like "isGlobal", "isTracker", "isStandAlone"?